src/components/SQForm/SQFormCheckboxGroup.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Grid} from '@material-ui/core';
+import {
+  SQFieldArray,
+  SQFormCheckboxGroupItem,
+  useSQFormContext
+} from '../../../src';
+
+export default function SQFormCheckboxGroup({
+  name,
+  children,
+  useSelectAll = false,
+  selectAllData = null,
+  selectAllContainerProps = {},
+  selectAllProps = {}
+}) {
+  const {values, setFieldValue} = useSQFormContext();
+
+  if (!useSelectAll) {
+    return <SQFieldArray name={name}>{children}</SQFieldArray>;
+  }
+
+  return (
+    <>
+      <Grid container {...selectAllContainerProps}>
+        <SQFormCheckboxGroupItem
+          name="selectAll"
+          label="Select All"
+          isChecked={values.selectAll}
+          onChange={e => {
+            if (e.target.checked) {
+              setFieldValue(name, selectAllData);
+            } else {
+              setFieldValue(name, []);
+            }
+            setFieldValue('selectAll', !values.selectAll);
+          }}
```

Comment:
For readability sake could we pull this anon function out to a const?

Reply:
sure! good idea.

Reply:
done